### PR TITLE
Merge `release-1.4.15` to `dev`

### LINKED ISSUE #3899: Merge back 1.4.15 and "make it work"
fix nightly tests and do any other needful things to merge this back to dev.

### PR DESCRIPTION
**Pull request by Fraser999**
This PR brings over the changes from `release-1.4.15` to `dev`.

The final commit fixes the issue of keeping the JSON-RPCs `chain_get_era_summary` and `chain_get_era_info_by_switch_block` functional for data before and after the 1.4.15 upgrade, where the `Key` variant for era summaries changed.

Tested locally with an nctl network upgraded from 1.4.13 to this code.

The PR also now includes an nctl test for this (upgrade_scenario_13.sh) proved by @alsrdn.

Closes #3899.

### CODE REVIEW

**Review by rafal-ch — APPROVED** [review]
LGTM :+1:

File: execution_engine_testing/test_support/src/lib.rs [hunk 1]
```diff
@@ -197,3 +195,45 @@ pub static PRODUCTION_ROUND_SEIGNIORAGE_RATE: Lazy<Ratio<u64>> = Lazy::new(|| {
...
+
+#[cfg(test)]
+mod tests {
+    use std::path::PathBuf;
+
+    use super::*;
+
+    #[test]
+    fn defaults_should_match_production_chainspec_values() {
+        let path = PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("resources/chainspec.toml");
```
Comment (mpapierski) [inline execution_engine_testing/test_support/src/lib.rs]: Why not `PRODUCTION_PATH` here? It has the same value

**Review by mpapierski — APPROVED** [review]
One comment on the constant, otherwise LGTM

File: execution_engine_testing/test_support/src/lib.rs (hunk 1 above)
Comment (Fraser999) [reply execution_engine_testing/test_support/src/lib.rs]: Good call.  Done in [f5b6158](https://github.com/casper-network/casper-node/pull/3918/commits/f5b61585e251861774e83409c578ab9f16b9a88a).

**Comment by alsrdn** [comment]
Also fixes https://github.com/casper-network/casper-node/issues/3896

**Pull request merged**